### ISSUE TITLE
Allow defining field names that don't follow the restrictions of python variables

### ISSUE THREAD

**Issue by dyeray** [scrapy/scrapy#3266]
Sometimes we may require defining field names that are not valid Python variables (for example containing spaces). I believe the correct way of doing this right now is that the developer needs to add a Pipeline that maps the internal name to the name we wan't to export. Scrapy could ship with a Pipeline (maybe enabled by default) that does this mapping for fields that define a field parameter. For example:

```
class MyItem(Item):
   date_from = Field(export_name='Initial date')
```

This is a very simple pipeline. However, since this is a very common use case, I think it is something that could be shipped by default. I'm happy to provide a PR for this.

**Comment by kmike** [comment]
If you're using dicts instead of scrapy.Item instances, you can do `yield {'Initial date': my_date_from, ...}`. 

But I like the idea, I think we can it built-in.

**Comment by VMRuiz** [comment]
I like that idea, but, wouldn't it be possible to reference the Item or the Item Loader by both the export_name or the var name instead of adding a new pipeline?

For example:

```
class MyItem(Item):
   date_from = Field(export_name='Initial date')

item = MyItem()
item['date_from'] = '01/01/1970' # This Works
item['Initial date'] = '01/01/1970' # This should work

loader = ItemLoader(item=MyItem()
loader.add_value('Initial date', '01/01/1970')  # This should work
```

**Comment by kmike** [comment]
https://github.com/scrapy/scrapy/pull/3696 also should solve this, though in a slightly different way.

**Comment by YngProgrammer** [comment]
Is there any PR to provide this by default as for now?
It could definitely improve the experience for many of us, rather than creating faulty pipelines and getting our hands dirty for no reason.

The "export_name" suggestion from @dyeray sounds like the perfect deal.

**Comment by kmike** [comment]
Regarding `export_name` idea, it's not solved by #3696. I wonder if we could make it a feature of https://github.com/scrapy/itemadapter.
